packaging/helm-charts/ksml/templates/servicemonitor.yaml

```diff
@@ -1,4 +1,4 @@
-{{- if and ( .Capabilities.APIVersions.Has "monitoring.coreos.com/v1" )  .Values.prometheus.enabled .Values.serviceMonitor.enabled }}
+{{- if and ( .Capabilities.APIVersions.Has "monitoring.coreos.com/v1" )  .Values.prometheus.enabled .Values.serviceMonitor.enabled  (eq "statefulset" (lower .Values.deploymentMode)) }}
```

Comment:
The service monitor can stay for jobs. Only prometheus rule is an issue as it falsely sends alerts for completed jobs.